### PR TITLE
Daniel/subscriptions/3.manage subscription

### PR DESCRIPTION
**Pull request by afterxleep**
Task/Issue URL: https://app.asana.com/0/72649045549333/1205054784245717/f

**Description**:
- Implement Subscription Management View
- Adds labels to the transaction progress box and reduces the opacity of the background view.

**Steps to test this PR**:
1. Purchase a Subscription
2. Observe that the Progress Dialog now tells you the current Status "Purchasing Subscription or Completing Purchase"
3. Go to Settings -> Manage Subscription
4. Tap `Change Plan or Billing` and observe you're taken to the iOS subscription management page
5. Tap `Remove From this device` and confirm
6. Try to purchase the subscription again.  It should be found and adequately restored
7. Observe that during Restore, the progress dialog says "Restoring subscription."

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/PrivacyPro/Views/PurchaseInProgressView.swift [hunk 1]
```diff
@@ -21,25 +21,42 @@ import SwiftUI
 struct PurchaseInProgressView: View {
     @Environment(\.colorScheme) var colorScheme
     
+    var status: String
+
```
Comment (miasma13) [inline DuckDuckGo/PrivacyPro/Views/PurchaseInProgressView.swift]: - Remaining TODO
- How about extracting numbers to constants?
- Is it possible to reuse colors from `DesignResourcesKit` ?

File: DuckDuckGo/SettingsViewModel.swift [hunk 2]
```diff
@@ -217,40 +213,64 @@ extension SettingsViewModel {
...
+        #endif
+        return SettingsState.NetworkProtection(enabled: enabled, status: "")
+    }
+    
+    private func getPrivacyProState() -> SettingsState.PrivacyPro {
+        var enabled = false
+        var canPurchase = false
+        let hasActiveSubscription = Self.cachedHasActiveSubscription
+        #if SUBSCRIPTION
+            enabled = featureFlagger.isFeatureOn(.privacyPro)
```
Comment (miasma13) [inline DuckDuckGo/SettingsViewModel.swift]: I know we already discussed the naming thing but still, for such concepts as a feature flag I would lean more towards using more neutral/generic descriptor and suggest using `.subscription`.

File: DuckDuckGo/PrivacyPro/Views/SubscriptionFlowView.swift [hunk 3]
```diff
@@ -26,21 +26,57 @@ struct SubscriptionFlowView: View {
...
         .onAppear(perform: {
             Task { await viewModel.initializeViewData() }
         })
         .navigationTitle(viewModel.viewTitle)
+        .navigationBarBackButtonHidden(viewModel.transactionStatus != .idle)
+        
+        // Active subscription found Alert
+        .alert(isPresented: $viewModel.hasActiveSubscription) {
+            Alert(
+                title: Text("Subscription Found"),
```
Comment (miasma13) [inline DuckDuckGo/PrivacyPro/Views/SubscriptionFlowView.swift]: Strings to be extracted.

File: DuckDuckGo/PrivacyPro/Views/SubscriptionSettingsView.swift [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                    SettingsCustomCell(content: {
+                        Text(UserText.privacyProChangePlan)
+                            .daxBodyRegular()
+                    },
+                                       action: { viewModel.manageSubscription() },
+                                       asLink: true)
+                }
+                Section(header: Text(UserText.privacyProHelpAndSupport),
+                        footer: Text(UserText.privacyProFAQFooter)) {
+                    NavigationLink(destination: Text("Privacy Pro FAQ")) {
```
Comment (miasma13) [inline DuckDuckGo/PrivacyPro/Views/SubscriptionSettingsView.swift]: String to be extracted.

**Review by miasma13 — APPROVED** [review]
Couple of minor comments and some subtasks left on https://app.asana.com/0/1203936086921904/1206337715257645/f

File: DuckDuckGo/SettingsViewModel.swift (hunk 2 above)
Comment (afterxleep) [reply DuckDuckGo/SettingsViewModel.swift]: Let's sync about this in the asana task.

Comment (afterxleep) [reply DuckDuckGo/SettingsViewModel.swift]: Everything has been renamed to "subscription"

**Comment by afterxleep** [comment]
@miasma13 Fixed all suggestions in Asana, and added link to load subscriptions via Storekit2.    Thanks for the feedback

**Pull request merged**